Source/ISHRemote/Trisoft.ISHRemote/Cmdlets/Annotation/AddIshAnnotation.cs

```diff
@@ -0,0 +1,345 @@
+
+        /// <summary>
+        /// <para type="description">The IshAnnotation array that needs to be created. Pipeline</para>
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = "IshAnnotationGroup")]
+        [ValidateNotNullOrEmpty]
+        public IshAnnotation[] IshAnnotation { get; set; }
+        
+        protected override void BeginProcessing()
+        {
+            if (IshSession == null) { IshSession = (IshSession)SessionState.PSVariable.GetValue(ISHRemoteSessionStateIshSession); }
+            if (IshSession == null) { throw new ArgumentException(ISHRemoteSessionStateIshSessionException); }
+            WriteDebug($"Using IshSession[{IshSession.Name}] from SessionState.{ISHRemoteSessionStateIshSession}");
+
+            if ((IshSession.ServerIshVersion.MajorVersion < 14) || ((IshSession.ServerIshVersion.MajorVersion == 14) && (IshSession.ServerIshVersion.RevisionVersion < 1)))
+            {
+                throw new PlatformNotSupportedException($"Add-IshAnnotation requires server-side Annotation API which is only available starting from 14.0.1 and up. ServerIshVersion[{IshSession.ServerVersion}]");
+            }
+            base.BeginProcessing();
+        }
+
+        /// <summary>
+        /// Process the Add-IshAnnotation commandlet.
+        /// </summary>
+        /// <exception cref="TrisoftAutomationException"></exception>
+        /// <exception cref="Exception"></exception>
+        /// <remarks>Writes <see cref="IshAnnotation"/> array to the pipeline</remarks>
+        protected override void ProcessRecord()
+        {
+            try
+            {
+                List<IshObject> returnedObjects = new List<IshObject>();
+
+                //1. Add annotations depending on the chosen ParameterSet
+                List<string> returnAnnotations = new List<string>();
+                IshFields returnFields = new IshFields();
+
+                if (ParameterSetName == "MetadataGroup")
+                {
+                    var metadata = IshSession.IshTypeFieldSetup.ToIshMetadataFields(ISHType, new IshFields(Metadata), Enumerations.ActionMode.Create);
+                    string annotationId = IshSession.Annotation25.Create(metadata.ToXml());
+                    returnAnnotations.Add(annotationId);
+                    returnFields = metadata;
+                }
+
+                if (ParameterSetName == "ParametersGroup" || ParameterSetName == "IshObjectGroup")
+                {
+                    // 1.1. Get the latest RevisionId for the given LogicalId/Version/Language of the content object
+                    var metadata = (Metadata == null) ? new IshFields() : new IshFields(Metadata);
+                    IshFields requestedMetadataContentObject = new IshFields();
+                    requestedMetadataContentObject.AddField(new IshRequestedMetadataField(FieldElements.ED, Enumerations.Level.Lng, Enumerations.ValueType.All));
```

Comment:
For performance reasons, I would select and set the element of the revisionId. So requesting the element of the ED field and setting the element of the FISHREVISIONID.